c_implementation/src/ak_fs.c

```
@@ -278,3 +278,94 @@ int ak_fs_ls()
+        char buffer[258] = {0};
+        size_t bytes_read = fread(&buffer, 1, sizeof(buffer), fd);
+        if ( bytes_read < sizeof(buffer) )
+        {
+            ak_log_error(__func__, "File is smaller than expected. Wrong format?");
+            fclose(fd);
+            return 2;
+        }
+        if ( buffer[128] != '\n' || buffer[257] != '\n' )
+        {
+            ak_log_error(__func__, "Unknown format");
+            return 2;
```

Cppcheck (2.10-2)

[error] Resource leak

(resourceLeak)